packages/mcp-server/scripts/generate-summaries.ts

```diff
@@ -0,0 +1,254 @@
+- "blog, content sites, markdown rendering"
+
+Requirements:
+- Maximum 200 characters (including spaces and commas)
+- Lowercase only
+- Comma-separated list of use cases
+- Focus on WHEN/WHY someone would need this, not what it is
+- Be specific about project types, features, or components when applicable
+- Use "always" sparingly, only for truly universal concepts
+- Do not include quotes or special formatting in your response
+- Respond with ONLY the use cases text, no additional text
+
+Here is the documentation page content to analyze:
+
+`;
+
+interface Section {
+	title: string;
+	use_cases: string;
+	slug: string;
+	url: string;
+}
+
+async function fetch_section_content(url: string): Promise<string> {
+	const response = await fetch(url, { signal: AbortSignal.timeout(30000) });
+	if (!response.ok) {
+		throw new Error(`Failed to fetch ${url}: ${response.status} ${response.statusText}`);
+	}
+	return await response.text();
+}
+
+async function main() {
+	console.log('🚀 Starting use cases generation...');
+
+	// Check for API key
+	const api_key = process.env.ANTHROPIC_API_KEY;
+	if (!api_key) {
+		console.error('❌ Error: ANTHROPIC_API_KEY environment variable is required');
+		console.error('Please set it in packages/mcp-server/.env file or export it:');
+		console.error('export ANTHROPIC_API_KEY=your_api_key_here');
+		process.exit(1);
+	}
+
+	// Get all sections
+	console.log('📚 Fetching documentation sections...');
+	let sections: Section[] = await get_sections();
+	console.log(`Found ${sections.length} sections`);
+
+	// Debug mode: limit to 2 sections
+	const debug_mode = process.env.DEBUG_MODE === '1';
+	if (debug_mode) {
+		console.log('🐛 DEBUG_MODE enabled - processing only 2 sections');
+		sections = sections.slice(0, 2);
+	}
+
+	// Fetch content for each section
+	console.log('📥 Downloading section content...');
+	const sections_with_content: Array<{
+		section: Section;
+		content: string;
+		index: number;
+	}> = [];
+	const download_errors: Array<{ section: string; error: string }> = [];
+
+	for (let i = 0; i < sections.length; i++) {
+		const section = sections[i]!;
+		try {
+			console.log(`Fetching ${i + 1}/${sections.length}: ${section.title}`);
+			const content = await fetch_section_content(section.url);
+			sections_with_content.push({
+				section,
+				content,
+				index: i,
+			});
+		} catch (error) {
+			const error_msg = error instanceof Error ? error.message : String(error);
+			console.error(`⚠️  Failed to fetch ${section.title}:`, error_msg);
+			download_errors.push({ section: section.title, error: error_msg });
+		}
+	}
+
+	console.log(`✅ Successfully downloaded ${sections_with_content.length} sections`);
+
+	if (sections_with_content.length === 0) {
+		console.error('❌ No sections were successfully downloaded');
+		process.exit(1);
+	}
+
+	// Initialize Anthropic client
+	console.log('🤖 Initializing Anthropic API...');
+	const anthropic = new AnthropicProvider('claude-sonnet-4-5-20250929', api_key);
```

Comment:
Do you think this could be a job for a cheaper model? It seems decently simple

Reply:
It costs ~$0.5 to do a full run via Anthropic Batch API, if we do it for only updated files it'll be pennies. I think once we agree on a style of use_cases (short descriptions, tags, etc) then we can see if it's reproducible on a cheaper model?

Reply:
Then I guess let's treat ourselves with premiumness 😄